pkg/gameserverallocations/metrics.go

```diff
@@ -32,23 +32,42 @@ import (
 )
 
 var (
+	logger = runtime.NewLoggerWithSource("metrics")
+
 	keyFleetName          = mt.MustTagKey("fleet_name")
 	keyClusterName        = mt.MustTagKey("cluster_name")
 	keyMultiCluster       = mt.MustTagKey("is_multicluster")
 	keyStatus             = mt.MustTagKey("status")
 	keySchedulingStrategy = mt.MustTagKey("scheduling_strategy")
 
-	gameServerAllocationsLatency = stats.Float64("gameserver_allocations/latency", "The duration of gameserver allocations", "s")
+	gameServerAllocationsLatency   = stats.Float64("gameserver_allocations/latency", "The duration of gameserver allocations", "s")
+	gameServerAllocationsErrorRate = stats.Int64("gameserver_allocations/errors", "The errors of gameserver allocations", "1")
 )
 
 func init() {
-	runtime.Must(view.Register(&view.View{
-		Name:        "gameserver_allocations_duration_seconds",
-		Measure:     gameServerAllocationsLatency,
-		Description: "The distribution of gameserver allocation requests latencies.",
-		Aggregation: view.Distribution(0, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2, 3),
-		TagKeys:     []tag.Key{keyFleetName, keyClusterName, keyMultiCluster, keyStatus, keySchedulingStrategy},
-	}))
+
+	stateViews := []*view.View{
+		{
+			Name:        "gameserver_allocations_duration_seconds",
+			Measure:     gameServerAllocationsLatency,
+			Description: "The distribution of gameserver allocation requests latencies.",
+			Aggregation: view.Distribution(0, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2, 3),
+			TagKeys:     []tag.Key{keyFleetName, keyClusterName, keyMultiCluster, keyStatus, keySchedulingStrategy},
+		},
+		{
+			Name:        "gameserver_allocations_retry_rate",
```

Comment:
```suggestion
			Name:        "gameserver_allocations_retry_total",
```

https://prometheus.io/docs/practices/naming/